src/components/app/data/hooks/useEnterpriseCustomer.test.jsx

```diff
@@ -1,53 +1,110 @@
 import { renderHook } from '@testing-library/react-hooks';
 import { QueryClientProvider } from '@tanstack/react-query';
 import { AppContext } from '@edx/frontend-platform/react';
+import { MemoryRouter, Route, Routes } from 'react-router-dom';
 import { authenticatedUserFactory, enterpriseCustomerFactory } from '../services/data/__factories__';
 import useEnterpriseCustomer from './useEnterpriseCustomer';
-import { queryClient } from '../../../../utils/tests';
-import { fetchEnterpriseLearnerData } from '../services';
+import { generateTestPermutations, queryClient } from '../../../../utils/tests';
+import { fetchEnterpriseLearnerDashboard, fetchEnterpriseLearnerData } from '../services';
 
 jest.mock('../services', () => ({
   ...jest.requireActual('../services'),
   fetchEnterpriseLearnerData: jest.fn().mockResolvedValue(null),
+  fetchEnterpriseLearnerDashboard: jest.fn().mockResolvedValue(null),
 }));
+
 const mockEnterpriseCustomer = enterpriseCustomerFactory();
 const mockAuthenticatedUser = authenticatedUserFactory();
 const mockEnterpriseLearnerData = {
   enterpriseCustomer: mockEnterpriseCustomer,
-  enterpriseCustomerUserRoleAssignments: [],
   activeEnterpriseCustomer: null,
   activeEnterpriseCustomerUserRoleAssignments: [],
   allLinkedEnterpriseCustomerUsers: [],
-  enterpriseFeatures: {},
   staffEnterpriseCustomer: null,
+  enterpriseFeatures: {
+    isBFFEnabled: false,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+};
+const mockBFFDashboardData = {
+  enterpriseCustomer: {
+    ...mockEnterpriseCustomer,
+    isBFFEnabled: true,
```

Comment:
nit: should `isBFFEnabled` be an attr under `enterpriseCustomer`? I don't think it's technically a property on an enterprise customer.

Reply:
This is mainly done in this and other test to have an attribute within enterprise features that is distinctive to not run into the case where comparing a enterprise learner API sourced enterprise feature is equivalent to the BFF sourced enterprise feature response object.

I went ahead and added a comment to highlight this along with updating the name of the feature to `isBFFSourcedAPIResponse`

Reply:
@brobro10000 I'm not sure we necessarily should have different responses between BFF vs. non-BFF, given `useEnterpriseCustomer`'s intent is to return the same data structure in its output regardless of the data source.

By having different data depending on the source, it raises some questions like, "do BFF vs. non-BFF actually have different data structures returned from this hook?"

Also, worth noting, I just tried commenting it out and re-run this test suite, and all the tests still pass without `enterpriseCustomer. isBFFEnabled`. Not sure if it's providing any utility as a result, if tests continue to pass without it?